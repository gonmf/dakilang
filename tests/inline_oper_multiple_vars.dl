test1(X, Res) :- eval(X, "($0 - 2) + 43 * (3 + $0)", Res).
test1(3, Res)?


test11(X, Y, Res) :- eval(X, Y, "$0 * ($1 + 2)", Res).
test11(3, 2, Res)?

test12(X, Y, Res) :- set(X * (Y + 2), Res).
test12(3, 2, Res)?

listing
