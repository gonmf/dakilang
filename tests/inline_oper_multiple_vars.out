> test1(X, Res) :- eval(X, "($0 - 2) + 43 * (3 + $0)", Res).
> test1(3, Res)?
Res = 264

>
>
> test11(X, Y, Res) :- eval(X, Y, "$0 * ($1 + 2)", Res).
> test11(3, 2, Res)?
Res = 12

>
> test12(X, Y, Res) :- set(X * (Y + 2), Res).
> test12(3, 2, Res)?
Res = 12

>
> listing
0: test1(X, Res) :- eval(X, '($0 - 2) + 43 * (3 + $0)', Res).
1: test11(X, Y, Res) :- eval(X, Y, '$0 * ($1 + 2)', Res).
2: test12(X, Y, Res) :- eval(X, Y, '$0*($1+2)', $1), set($1, Res).

